File: backend/src/main/java/com/team05/petmeeting/domain/feed/dto/FeedListRes.java
package com.team05.petmeeting.domain.feed.dto;

import com.team05.petmeeting.domain.feed.entity.Feed;
import com.team05.petmeeting.domain.feed.enums.FeedCategory;
import java.time.LocalDateTime;

public record FeedListRes(
        Long feedId,
        Long userId,
        String profileImageUrl,
        Long animalId,
        String nickname,
        FeedCategory category,
        String title,
        String content,
        String imageUrl,
        int likeCount,
        boolean isLiked,
        int commentCount,
        LocalDateTime createdAt,
        LocalDateTime updatedAt
) {
    public FeedListRes(
            Feed feed,
            Long userId,
            String profileImageUrl,
            String nickname,
            Long animalId,
            Long likeCount,
            Long commentCount,
            boolean isLiked
    ) {
        this(
                feed.getId(),
                userId,
                profileImageUrl,
                animalId,
                nickname,
                feed.getCategory(),
                feed.getTitle(),
                feed.getContent(),
                feed.getImageUrl(),
                likeCount.intValue(),
                isLiked,
                commentCount.intValue(),
                feed.getCreatedAt(),
                feed.getUpdatedAt()
        );
    }
}

File: backend/src/main/java/com/team05/petmeeting/domain/feed/dto/FeedListRes.kt
package com.team05.petmeeting.domain.feed.dto

import com.team05.petmeeting.domain.feed.entity.Feed
import com.team05.petmeeting.domain.feed.enums.FeedCategory
import java.time.LocalDateTime

data class FeedListRes(
    val feedId: Long,
    val userId: Long,
    val profileImageUrl: String?,
    val animalId: Long?,
    val nickname: String,
    val category: FeedCategory,
    val title: String,
    val content: String,
    val imageUrl: String?,
    val likeCount: Int,
    val isLiked: Boolean,
    val commentCount: Int,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime?
) {
    companion object {
        fun from(
            feed: Feed,
            userId: Long,
            profileImageUrl: String?,
            nickname: String,
            animalId: Long?,
            likeCount: Long,
            commentCount: Long,
            isLiked: Boolean
        ) = FeedListRes(
            feed.id,
            userId,
            profileImageUrl,
            animalId,
            nickname,
            feed.category,
            feed.title,
            feed.content,
            feed.imageUrl,
            likeCount.toInt(),
            isLiked,
            commentCount.toInt(),
            feed.createdAt,
            feed.updatedAt
        )
    }
}